spooq/transformer/null_cleaner.py

```diff
@@ -0,0 +1,168 @@
+     |    |-- key: string
+     |    |-- value: string (valueContainsNull = true)
+
+    Parameters
+    ----------
+    thresholds : :py:class:`dict`
+        Dictionary containing column names and respective valid ranges
+
+    column_to_log_cleansed_values : :any:`str`, Defaults to None
+        Defines a column in which the original (uncleansed) value will be stored in case of cleansing. If no column
+        name is given, nothing will be logged.
+
+    store_as_map : :any:`bool`, Defaults to False
+        Specifies if the logged cleansed values should be stored in a column as :any:`pyspark.sql.types.MapType` with
+        stringified values or as :any:`pyspark.sql.types.StructType` with the original respective data types.
+
+    Note
+    ----
+    Following cleansing rule attributes per column are supported:
+
+        * min, mandatory - any
+            A number or timestamp/date which serves as the lower limit for allowed values. Values
+            below this threshold will be cleansed.
+        * max, mandatory - any
+            A number or timestamp/date which serves as the upper limit for allowed values. Values
+            above this threshold will be cleansed.
+        * default, defaults to None - :class:`~pyspark.sql.column.Column` or any primitive Python value
+            If a value gets cleansed it gets replaced with the provided default value.
+
+    The :meth:`pyspark.sql.functions.between` method is used internally.
+
+    Returns
+    -------
+    :any:`pyspark.sql.DataFrame`
+        The transformed DataFrame
+
+    Raises
+    ------
+    :any:`exceptions.ValueError`
+        Threshold-based cleaning only supports Numeric, Date and Timestamp Types!
+        Column with name: {col_name} and type of: {col_type} was provided
+
+    Warning
+    -------
+    Only Numeric, TimestampType, and DateType data types are supported!
+    """
+
+    def __init__(self, cleaning_definitions={}, column_to_log_cleansed_values=None, store_as_map=False):
+        super().__init__(
+            cleaning_definitions=cleaning_definitions,
+            column_to_log_cleansed_values=column_to_log_cleansed_values,
+            store_as_map=store_as_map,
+            temporary_columns_prefix="dac28b56d8055953a7038bfe3b5097e7",
+        )
+        self.logger.debug("Range Definitions: " + str(self.cleaning_definitions))
+
+    def transform(self, input_df):
+        self.logger.debug("input_df Schema: " + input_df._jdf.schema().treeString())
+        ordered_column_names = input_df.columns
+
+        if self.column_to_log_cleansed_values:
+            column_names_to_clean = self.cleaning_definitions.keys()
+            temporary_column_names = self._get_temporary_column_names(column_names_to_clean)
+            input_df = self._add_temporary_columns(input_df, column_names_to_clean, temporary_column_names)
+
+        cleansing_expressions = []
+        for column_name, value_range in list(self.cleaning_definitions.items()):
```

Comment:
not really a value_range. Maybe just "cleansing_definition" or something like that

Reply:
Renamed.